spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/GCPProperties.java

```diff
@@ -27,7 +28,7 @@
 
 	private String projectId;
 
-	private String jsonKeyFile;
+	private Resource credentialKeyFile;
```

Comment:
They should not be getting credentials from the Properties. In boot, the context auto configuration should provide the credentials.

Now, how does that object get created? Either via defaults or, if someone set the path to the credentials file, it comes from that file.